### PR TITLE
feat(nuxt): add `workspaceDir` option and add it to `modulesDir`

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves  #158

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This adds `workspaceDir` as an option in nuxt config. By default it uses `pkg-types` to locate this directory.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/config/_common.ts [hunk 1]
```diff
@@ -51,6 +51,19 @@ export default defineUntypedSchema({
...
+   * Define the workspace directory of your application.
+   *
+   * Often this is used when in a monorepo setup. Nuxt will attempt to detect
+   * your workspace directory automatically, but you can override it here.
+   *
+   * It is normally not needed to configure this option.
+   * @version 3
+   */
+  workspaceDir: {
+    $resolve: (val, get) => val ? resolve(get('rootDir'), val) : ''
```
Comment (pi0) [inline packages/schema/src/config/_common.ts:64]: It would be nice if we directly use `fineWorkspaceDir` here for fallback value instead of core

Comment (danielroe) [reply packages/schema/src/config/_common.ts:64]: Ah, but it's async.

Comment (pi0) [reply packages/schema/src/config/_common.ts:64]: On it!

Comment (pi0) [reply packages/schema/src/config/_common.ts:64]: Should be possible now

**Comment by pi0** [comment]
Would be nice if in same PR, we can sync workpaceDir to the nitro config as well (releasing next version shortly)

**Comment by danielroe** [comment]
Great! Will update shortly. I think I already added the option in Nitro config in this PR (though it won't be effective until next Nitro version).

**Pull request merged**